subsys/bluetooth/controller/ll_sw/ull_conn_iso.c

```diff
@@ -304,37 +264,65 @@ void ull_conn_iso_done(struct node_rx_event_done *done)
 	}
 }
 
-static void ticker_resume_op_cb(uint32_t status, void *param)
+/**
+ * @brief Stop and tear down a connected ISO stream
+ * This function may be called to tear down a CIS. When the CIS teardown
+ * has completed and the stream is released and callback is provided, the
+ * cis_released_cb callback is invoked.
+ *
+ * @param cis		 Pointer to connected ISO stream to stop
+ * @param cis_relased_cb Callback to invoke when the CIS has been released.
+ *                       NULL to ignore.
+ */
+void ull_conn_iso_cis_stop(struct ll_conn_iso_stream *cis,
+			   ll_iso_stream_released_cb_t cis_released_cb)
 {
-	ARG_UNUSED(param);
-
-	LL_ASSERT(status == TICKER_STATUS_SUCCESS);
-}
+	struct ll_conn_iso_group *cig;
+	struct ull_hdr *hdr;
 
-static void ticker_resume_cb(uint32_t ticks_at_expire, uint32_t remainder,
-			     uint16_t lazy, uint8_t force, void *param)
-{
-	static memq_link_t link;
-	static struct mayfly mfy = {0, 0, &link, NULL, lll_resume};
-	struct lll_event *resume_event;
-	uint32_t ret;
+	if (cis->teardown) {
+		/* Teardown already started */
+		return;
+	}
+	cis->teardown = 1;
+	cis->released_cb = cis_released_cb;
 
-	LL_ASSERT(lazy == 0);
+	/* Check ref count to determine if any pending LLL events in pipeline */
+	cig = cis->group;
+	hdr = &cig->ull;
+	if (ull_ref_get(hdr)) {
+		static memq_link_t link;
+		static struct mayfly mfy = {0, 0, &link, NULL, lll_disable};
```

Comment:
Yes, this is correct. For some reason the calling context was assumed to be ULL_LOW, but it's called from conn_cleanup, which is ULL_HIGH.